activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -305,7 +305,8 @@ def migrate_status
       end
 
       def check_target_version
-        if target_version && !(Migration::MigrationFilenameRegexp.match?(ENV["VERSION"]) || /\A\d+\z/.match?(ENV["VERSION"]))
+        if target_version && !(Migration::MigrationFilenameRegexp.match?(ENV["VERSION"]) \
+                               || /\A(?=\d)[_\d]+(?<=\d)\z/.match?(ENV["VERSION"]))
```

Comment:
Seems like it would be simpler to update `MigrationFilenameRegexp` rather than to add an ad hoc fix here which is brittle.

Reply:
Would it be a big deal if we allowed underscore in the migration filenames? Would seem fine to me.

Reply:
Would it change the number of someone's existing `1_1_foo.rb` file?

Reply:
Ah, yes it would, not sure if `1_1_foo.rb` would be a working file though, as a constant would be inferred from it, and that would be an invalid constant name. So I think we're good in term of backward compatibility, but I may be missing something.